src/mono/msbuild/android/build/AndroidApp.targets

```diff
@@ -85,6 +85,8 @@
       <MonoAOTCompilerDefaultAotArguments Include="static" />
       <MonoAOTCompilerDefaultAotArguments Include="dwarfdebug" />
       <MonoAOTCompilerDefaultAotArguments Condition="'$(_IsLibraryMode)' == 'true'" Include="direct-icalls" />
+      <MonoAOTCompilerDefaultAotArguments Condition="'$(_IsLibraryMode)' == 'true' and $(CustomRuntimeInitCallback) == ''" Include="runtime-init-callback" />
```

Comment:
I think it would make sense to have the option to opt out of runtime-init-callback and handle of of the runtime init when building a static library. Since we will hide all symbols for shared library we need to do the runtime-init-callback, so now way to opt out of it in that mode, but for static library we should have the option.

Reply:
Added an option `UsesRuntimeInitCallback` that users can set to not `true` in order to omit any `runtime-init-callback*` args being passed to the mono compiler.

So `UsesRuntimeInitCallback` and `CustomRuntimeInitCallback` can be used in conjunction to let the compiler know whether or not a `runtime-init-callback` is needed, and if it is, whether or not it should be the default or the custom passed in via `CustomRuntimeInitCallback`. Not sure what the best way right now to surface those properties to the public. I suppose once we have a doc to demonstrate how all of the piece fit together, we can surface it there?
```
    <PropertyGroup Condition="'($_IsLibraryMode)' == 'true'">
      <_UsesRuntimeInitCallback>$(UsesRuntimeInitCallback)</_UsesRuntimeInitCallback>
      <_UsesRuntimeInitCallback Condition="'$(_UsesRuntimeInitCallback)' == ''">true</_UsesRuntimeInitCallback>
    </PropertyGroup>
    
    <ItemGroup>
      ...
      <MonoAOTCompilerDefaultAotArguments Condition="'$(_UsesRuntimeInitCallback)' == 'true' and $(CustomRuntimeInitCallback) == ''" Include="runtime-init-callback" />
      <MonoAOTCompilerDefaultAotArguments Condition="'$(_UsesRuntimeInitCallback)' == 'true' and $(CustomRuntimeInitCallback) != ''" Include="runtime-init-callback=$(CustomRuntimeInitCallback)" />
    </ItemGroup>
```

Reply:
Yes we surface them when needed, by default it should be using runtime init callback and it should use our default implementation. Turning off UsesRuntimeInitCallback should also omit creating autoinit.c since that won't be needed.